pkg/sdn/apis/network/validation/validation.go

```diff
@@ -25,36 +25,43 @@ func SetDefaultClusterNetwork(cn sdnapi.ClusterNetwork) {
+			} else if cn.HostSubnetLength < 2 {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("hostSubnetLength"), cn.HostSubnetLength, "subnet length must be at least 2"))
+			}
+		}
 
-	if (clusterIPNet != nil) && (serviceIPNet != nil) && clusterIPNet.Contains(serviceIPNet.IP) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, "service network overlaps with cluster network"))
-	}
-	if (serviceIPNet != nil) && (clusterIPNet != nil) && serviceIPNet.Contains(clusterIPNet.IP) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("network"), clusterNet.Network, "cluster network overlaps with service network"))
+		serviceIPNet, err := netutils.ParseCIDRMask(clusterNet.ServiceNetwork)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, err.Error()))
+		}
+		if (clusterIPNet != nil) && (serviceIPNet != nil) && clusterIPNet.Contains(serviceIPNet.IP) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, "service network overlaps with cluster network"))
+		}
+		if (serviceIPNet != nil) && (clusterIPNet != nil) && serviceIPNet.Contains(clusterIPNet.IP) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("network"), clusterNet.Network, "cluster network overlaps with service network"))
+		}
 	}
 
 	if clusterNet.Name == sdnapi.ClusterNetworkDefault && defaultClusterNetwork != nil {
-		if clusterNet.Network != defaultClusterNetwork.Network {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("network"), clusterNet.Network, "cannot change the default ClusterNetwork record via API."))
-		}
-		if clusterNet.HostSubnetLength != defaultClusterNetwork.HostSubnetLength {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("hostSubnetLength"), clusterNet.HostSubnetLength, "cannot change the default ClusterNetwork record via API."))
+		if len(clusterNet.ClusterNetworks) == len(defaultClusterNetwork.ClusterNetworks) {
+			for index := range defaultClusterNetwork.ClusterNetworks {
+				if clusterNet.ClusterNetworks[index].CIDR != defaultClusterNetwork.ClusterNetworks[index].CIDR || clusterNet.ClusterNetworks[index].HostSubnetLength != defaultClusterNetwork.ClusterNetworks[index].HostSubnetLength {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("ClusterNetworks"), clusterNet.ClusterNetworks, "cannot change the default ClusterNetwork record via API"))
+				}
+			}
+		} else {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("ClusterNetworks"), clusterNet.ClusterNetworks, "cannot change the default ClusterNetwork record via API"))
 		}
 		if clusterNet.ServiceNetwork != defaultClusterNetwork.ServiceNetwork {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, "cannot change the default ClusterNetwork record via API."))
```

Comment:
All these validations are between clusterNetwork and other resources like serviceNetwork, etc. but we should also check overlap among the multiple clusterNetwork cidrs.

Reply:
@pravisankar This condition is checked elsewhere... in parseNetworkInfo() (pkg/sdn/plugin/common.go). I thought it was more appropriate there as the clusterNetwork stucts are being created. Should it be moved?